test/compiler_test.exs

```diff
@@ -3857,4 +3858,37 @@ defmodule CompilerTest do
              }
     end
   end
+
+  describe "hooks" do
+    test "initialize hook" do
+      model =
+        Axon.input({nil, 1})
+        |> Axon.dense(1, kernel_initializer: :ones)
+        |> Axon.attach_hook(&IO.inspect/1, on: :initialize)
```

Comment:
For testing, use something like `send(self(), :from_hook)` with `assert_received`. :)